src/devices/bus/spectrum/neogs.cpp

```diff
@@ -90,27 +104,44 @@ class neogs_device : public device_t, public device_zxbus_card_interface
 	memory_bank_creator m_bank_rom;
 	memory_bank_creator m_bank_ram;
 	memory_view m_view;
-	required_device_array<dac_word_interface, 8> m_dac;
+	required_device_array<dac_word_interface, 2> m_dac;
+	required_device<spi_sdcard_sdhc_device> m_sdcard;
 
 private:
+	TIMER_CALLBACK_MEMBER(spi_clock);
+	emu_timer *m_spi_clock;
+	int m_spi_clock_cycles;
+	bool m_spi_clock_state;
+
 	template <u8 Bank> u8 ram_bank_r(offs_t offset);
 	template <u8 Bank> void ram_bank_w(offs_t offset, u8 data);
 
-	u8 status_r() { return m_status; }
-	void command_w(u8 data) { m_status |= 0x01; m_command_in = data; }
-	u8 data_r() { m_status &= ~0x80; return m_data_out; }
-	void data_w(u8 data) { m_status |= 0x80; m_data_in = data; }
+	u8 status_r() { LOGSTATUS(" read STAT & %02X\n", m_status); return m_status | 0x7e; }
+	void command_w(u8 data) { m_status |= 0x01; LOGSTATUS("write CMD & %02X, status: %02X\n", data, m_status); m_command_in = data; }
+	u8 data_r() { m_status &= ~0x80; LOGSTATUS(" read DATA & %02X, status: %02X\n", m_data_out, m_status); return m_data_out; }
+	void data_w(u8 data) { m_status |= 0x80; LOGSTATUS("write: DATA & %02X, status: %02X\n", data, m_status); m_data_in = data; }
```

Comment:
You need to check `machine().side_effects_disabled()` on the read with side effects (sorry, missed this one last time).  It’s also generally a good idea to check it before logging reads to avoid log spam when debugger windows are open.